src/components/Icon/svgIcons.tsx

```diff
@@ -0,0 +1,17 @@
+import * as React from 'react'
+
+export interface ISvgIconSpec {
+  element: any
+  viewBox?: string
+}
+
+const svgIcons: { [name: string]: ISvgIconSpec } = {
+  umbrella: {
+    viewBox: '0 0 34 34',
+    element: (
+      <path d="M27 14h5c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2v0zM27 14c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2v0 14c0 1.112-0.895 2-2 2-1.112 0-2-0.896-2-2.001v-1.494c0-0.291 0.224-0.505 0.5-0.505 0.268 0 0.5 0.226 0.5 0.505v1.505c0 0.547 0.444 0.991 1 0.991 0.552 0 1-0.451 1-0.991v-14.009c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2c0-5.415 6.671-9.825 15-9.995v-1.506c0-0.283 0.224-0.499 0.5-0.499 0.268 0 0.5 0.224 0.5 0.499v1.506c8.329 0.17 15 4.58 15 9.995h-5z" />
```

Comment:
A lot of SVGs can get really long and be pretty complex. I wonder if we might want to consider not inlining them like this in the code as I think it could get really hard to scan and edit. In our project today, an SVG has its own html file which gives the added benefit of formatting

Reply:
yes, there multiple ways to embrace this use case  - essentially, there is no difference for the common Icon's logic on where and how the child content will be provided. There are several HTML-to-JSX transpilers that could be introduced as part of the build logic - those will consume HTML templates as input and create corresponding JS modules for each of the icons. 

At the same time, the same could be achieved much simplier - if I've understood you correctly, the situation is that there is a dedicated HTML template for each SVG icon. In that case `svgIcons` module could be organised in the following way:

```jsx
// teamCreateIcon.tsx - provides 'teams-create' icon, only
const teamCreateIcon = (
   // HTML template goes here - essentially, the same syntax
  <g>
     <path d="..."/>
      ...
  </g>
)

export default teamCreateIcon
```

```js
// index.js - just assembling all the definitions from multiple files
export { default as TeamCreateIcon } from './teamCreateIcon'
export { default as StartCallIcon } from './startCallIcon'
```